### PR TITLE
Add code formatting guidelines

### PR DESCRIPTION
**Pull request by amulware**
Please review and check that all rules are agreed upon, and that all examples make sense.

### CODE REVIEW

File: docs/style-guide.md [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+
+// Don't
+private void Foo()
+{
+}
+private int Baz => 0;
+```
+
+Indentation is done with four spaces
+All public and protected members use PascalCase
```
Comment (willsb) [inline docs/style-guide.md]: 🔥  this line

Comment (amulware) [inline docs/style-guide.md]: ©️🍝

**Comment by amulware** [comment]
What aspects of lambdas do you mean specifically? And do you have a proposal?

**Comment by willsb** [comment]
Something like:

```csharp

//Do
SomeMethodThatConsumesALambda(() =>
{
    //Content goes here
});

//Don't
SomeMethodThatConsumesALambda(() => {
    //Content goes here
});
```

I would also like us to agree on `_` for ignored lambda parameters:
```csharp

//Do
SomeMethodThatConsumesALambda(_ => true);

//Don't
SomeMethodThatConsumesALambda(ignored => true);
```

Last but not least, I forgot ask: please add proper [syntax highlighting](https://help.github.com/articles/creating-and-highlighting-code-blocks/) to code blocks.

**Comment by amulware** [comment]
I'm not sure if I agree with the lambda syntax, though it's not a huge thing for me, and also depends a lot on the length of the method, and parameters, etc.

I do something like the following at times.
```
SomeMethodThatConsumesALambda(
    () => {
        //Content goes here
    });

// or maybe even

SomeMethodThatConsumesALambda(
    () => {
        //Content goes here
    }
);
```

Ah well. I'm fine with using `_`, though what about multiple parameters? `__, ___`? I tend to just use single letter names and plain not use them. But I'm mostly indifferent.

**Comment by willsb** [comment]
Agreed

**Pull request merged**